webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/SQLCodeEditorPanel.tsx

```diff
@@ -112,6 +112,64 @@ export const SQLCodeEditorPanel: TabContainerPanelComponent<ISqlEditorModeProps>
     ],
   });
 
+  useExecutor({
+    executor: data.onFormat,
+    handlers: [
+      function setCursor([script, formatted]) {
+        const view = editorRef?.view;
+
+        if (!view) {
+          return;
+        }
+
+        const selection = view.state.selection.main;
+
+        setTimeout(() => {
+          view.dispatch({
+            selection: {
+              anchor: selection.from,
+              head: selection.from,
+            },
+          });
+        });
+      },
+    ],
+  });
+
+  useExecutor({
+    executor: data.dataSource?.onSetScript,
+    postHandlers: [
+      function setScript({ script, source }) {
+        if (source === 'insert') {
+          // Note: Use setTimeout to ensure this runs after the current transaction is fully processed
```

Comment:
I don't really understand why `setTimeout` is here. Is this cause you want to onSetScript being executed first, and then executes what is inside `setTimeout`, right?

If so, I believe it is better to go with `next` props of `useExecutor`. It should be something like this in order to have more expected behavior from EventLoop execution perspective:

```
const setScriptExecutor = new SyncExecutor<ISetScriptData>();

  useExecutor({
    executor: data.dataSource?.onSetScript,
    next: setScriptExecutor,
  });

  useExecutor({
    executor: setScriptExecutor,
    handlers: [
      function setScript({ script, source }) {
        if (source === 'insert') {
          // Note: Use setTimeout to ensure this runs after the current transaction is fully processed
          const view = editorRef?.view;

          if (!view) {
            return;
          }

          const currentContent = view.state.doc.toString();
          const startPos = currentContent.lastIndexOf(script);

          if (startPos === -1) {
            return;
          }

          const endPos = startPos + script.length;

          view.dispatch({
            selection: {
              anchor: endPos,
              head: endPos,
            },
          });
        }
      },
    ],
  });
```